Style: Partially apply clang-tidy's `cppcoreguidelines-pro-type-member-init`

akien-mga opened the pull request:
Didn't commit all the changes where it wants to initialize a struct
with `{}`. Should be reviewed in a separate PR.

Option `IgnoreArrays` enabled for now to be conservative, can be
disabled to see if it proposes more useful changes.

Also fixed manually a handful of other missing initializations / moved
some from constructors.
(The `clang-tidy` check only handles the members which did *not* have an explicit initialization in the constructor.)

bruvzg commented on modules/text_server_adv/text_server_adv.cpp:
```diff
@@ -878,8 +878,8 @@ _FORCE_INLINE_ TextServerAdvanced::FontTexturePosition TextServerAdvanced::find_
 	msdfgen::Point2 position;
-	msdfgen::Shape *shape;
-	msdfgen::Contour *contour;
+	msdfgen::Shape *shape = nullptr;
+	msdfgen::Contour *contour = nullptr;
```
The same is also used in the `text_server_fb`.

akien-mga replied:
Thanks, fixed.

akien-mga committed:
Style: Partially apply clang-tidy's `cppcoreguidelines-pro-type-member-init`

Didn't commit all the changes where it wants to initialize a struct
with `{}`. Should be reviewed in a separate PR.

Option `IgnoreArrays` enabled for now to be conservative, can be
disabled to see if it proposes more useful changes.

Also fixed manually a handful of other missing initializations / moved
some from constructors.
